daemon/server/router/container/container_routes.go

```diff
@@ -485,26 +497,14 @@ func (c *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 
 	name := r.Form.Get("name")
 
-	config, hostConfig, networkingConfig, err := c.decoder.DecodeConfig(r.Body)
+	// TODO(thaJeztah): do we prefer [backend.ContainerCreateConfig] here?
```

Comment:
Probably should split this in two;

- A regular (through httputils.ReadJSON) unmarshal of the container.CreateRequesr
- A function to produce a backend.CreateConfig, which also validates the basics and sets the defaults.